lighthouse-core/report/v2/renderer/dom.js

```diff
@@ -37,7 +37,9 @@ class DOM {
       element.className = className;
     }
     Object.keys(attrs).forEach(key => {
-      element.setAttribute(key, attrs[key]);
+      if (attrs[key] !== undefined) {
```

Comment:
> Every caller will have to do that. I'd rather just prevent attr="undefined" in one place. Kinda what templating languages do for you ;)

I'd imagine most templating languages would give you `title="undefined"` in this case (which I assume is why the [current partial](https://github.com/GoogleChrome/lighthouse/blob/605a4579c5cd0eeba9ea71a3dcf38d7817d1b86a/lighthouse-core/report/partials/cards.html#L5) has `{{#if snippet}}title="{{snippet}}"{{/if}}` in it). I'd generally consider accidentally passing in attributes with `undefined` values a bug and it would be nice to catch that